### PR TITLE
17 - Mailer

### PR DESCRIPTION
**Pull request by mar7in91Wolox**
# Summary 
- I added the Mailer class.
- I added the Send Movie Info feature into the app. 

# Trello Card
https://trello.com/c/5XIjII4T

# Screenshots
image
image
image

### CODE REVIEW

File: training-net/src/Startup.cs [hunk 1]
```diff
@@ -12,10 +11,11 @@
 using training_net.Models;
 using training_net.Repositories.Database;
 using training_net.Repositories.Interfaces;
+using training_net.Mail;
 
 namespace training_net
 {
-  public class Startup
+    public class Startup
```
Comment (truccomarcos) [inline training-net/src/Startup.cs]: Is this indentation ok?

File: training-net/src/Views/Movie/SendMovieInfo.cshtml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+@{ViewData["DetailsDeleteTitle"] = Localizer["SendMovieInfoTitle"].Value;}
+<partial name="~/Views/Shared/_Details.cshtml" view-data="ViewData"  />
+<form asp-controller="Movies" asp-action="Index" method="post">
+    <p>
+        @Localizer["SendTo"]: <input type="text" name="EmailAddress">
+        <input type="submit" value="@Localizer["Send"]" />
+    </p>
+</form>
+<div>
+<a asp-action="Index">@Localizer["BackToList"].Value</a>
```
Comment (truccomarcos) [inline training-net/src/Views/Movie/SendMovieInfo.cshtml]: Don't forget to add indentation here too

File: training-net/src/Controllers/AccountController.cs [hunk 3]
```diff
@@ -47,7 +48,11 @@ public async Task<IActionResult> Login(LoginViewModel loginViewModel)
             if(ModelState.IsValid)
             {
                 var result = await SignInManager.PasswordSignInAsync(loginViewModel.UserName, loginViewModel.Password, loginViewModel.RememberMe, false);
-                if(result.Succeeded) return RedirectToAction("Users", "UserManagement");
+                if(result.Succeeded) 
+                {
+                    Mailer.Send("martin.perez@wolox.com.ar", "Login", "Te logueaste amiguito");
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/AccountController.cs]: Sorry!

File: training-net/src/Controllers/MovieController.cs [hunk 4]
```diff
@@ -189,5 +190,54 @@ public IActionResult DeleteConfirmed([FromForm] int? id)
...
+        public IActionResult SendMovieInfo([FromForm] string EmailAddress, int? id)
+        {
+            try
+            {
+                if (id == null)
+                    throw new NullReferenceException();
+                var movie = UnitOfWork.MovieRepository.Get(id.Value);
+                if (movie == null)
+                    return NotFound();
+                string bodyMsg =    $"Title: {movie.Title}{Environment.NewLine}"+
```
[the remark this answers is not in the document]
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Thanks!

File: training-net/src/Startup.cs (hunk 1 above)
Comment (mar7in91Wolox) [reply training-net/src/Startup.cs]: Done!

File: training-net/src/Views/Movie/SendMovieInfo.cshtml (hunk 2 above)
Comment (mar7in91Wolox) [reply training-net/src/Views/Movie/SendMovieInfo.cshtml]: Done!

File: training-net/src/Controllers/MovieController.cs [hunk 5]
```diff
@@ -189,5 +190,56 @@ public IActionResult DeleteConfirmed([FromForm] int? id)
...
+        public IActionResult SendMovieInfo(int? id)
+        {
+            try
+            {
+                if (id == null)
+                    throw new NullReferenceException();
+                MovieViewModel movieVM = new MovieViewModel();
+                var movie = UnitOfWork.MovieRepository.Get(id.Value);
+                if (movie == null)
+                    return NotFound();
```
Comment (truccomarcos) [inline training-net/src/Controllers/MovieController.cs]: Don't forget to handle this as an exception

File: training-net/src/Controllers/MovieController.cs [hunk 6]
```diff
@@ -189,5 +190,56 @@ public IActionResult DeleteConfirmed([FromForm] int? id)
...
+        [HttpPost("SendMovieInfo")]
+        public IActionResult SendMovieInfo([FromForm] string EmailAddress, int? id)
+        {
+            try
+            {
+                if (id == null)
+                    throw new NullReferenceException();
+                var movie = UnitOfWork.MovieRepository.Get(id.Value);
+                if (movie == null)
+                    return NotFound();
```
Comment (truccomarcos) [inline training-net/src/Controllers/MovieController.cs]: Same here

File: training-net/src/Controllers/MovieController.cs (hunk 5 above)
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Ups!

File: training-net/src/Controllers/MovieController.cs (hunk 6 above)
Comment (mar7in91Wolox) [reply training-net/src/Controllers/MovieController.cs]: Ups! x 2

**Pull request merged**